connector-service-impl/src/test/java/org/interledger/connector/links/DefaultNextHopPacketMapperTest.java

```diff
@@ -1,39 +1,259 @@
+
+	@Mock
+	private InterledgerAddressUtils addressUtils;
+
+	@Mock
+	private JavaMoneyUtils javaMoneyUtils;
+
+	@Mock
+	private AccountSettingsLoadingCache accountSettingsLoadingCache;
+
+	@Before
+	public void setup() {
+		defaultNextHopPacketMapper = new DefaultNextHopPacketMapper(connectorSettingsSupplier,
+				externalRoutingService, addressUtils, javaMoneyUtils, accountSettingsLoadingCache);
+	}
+
+	@Test
+	public void getNextHopPacket() {
+		final AccountId accountId = AccountId.of(UUID.randomUUID().toString());
+		final AccountId accountIdNextHopRoute = AccountId.of(UUID.randomUUID().toString());
+		final AccountSettings sourceAccountSettings = buildAccountSettings(accountId)
+				.build();
+		final AccountSettings destinationAccountSettings = buildAccountSettings(accountId)
+				.build();
+		final InterledgerPreparePacket sourcePacket = buildInterLedgerPreparePacket(10);
+		final Optional<Route> optionalRoute = Optional.of(Route.builder()
+				.nextHopAccountId(accountIdNextHopRoute)
+				.routePrefix(DEFAULT_TARGET_ADDRESS_PREFIX)
+				.build());
+		doReturn(optionalRoute).when(externalRoutingService).findBestNexHop(sourcePacket.getDestination());
+		when(accountSettingsLoadingCache.safeGetAccountId(eq(optionalRoute.get().nextHopAccountId()))).thenReturn(destinationAccountSettings);
+		when(addressUtils.isExternalForwardingAllowed(sourcePacket.getDestination())).thenReturn(false);
+
+		final NextHopInfo nextHopInfo = defaultNextHopPacketMapper.getNextHopPacket(sourceAccountSettings, sourcePacket);
+
+		assertThat(nextHopInfo.nextHopAccountId()).isEqualTo(optionalRoute.get().nextHopAccountId());
+		assertThat(nextHopInfo.nextHopPacket().getAmount().intValue()).isEqualTo(10);
+		assertThat(nextHopInfo.nextHopPacket().getDestination().getValue()).isEqualTo(TEST_ADDRESS_2);
+		assertThat(nextHopInfo.nextHopPacket().getExpiresAt()).isNotNull();
+	}
+
+	@Test(expected = InterledgerProtocolException.class)
+	public void getNextHopPacketNoDestinationAddressFoundFromRoutingService() {
+		final AccountId accountId = AccountId.of(UUID.randomUUID().toString());
+		final AccountSettings sourceAccountSettings = buildAccountSettings(accountId)
+				.build();
+		final InterledgerPreparePacket sourcePacket = buildInterLedgerPreparePacket(10);
+		final Optional<Route> optionalRoute = Optional.empty();
+		doReturn(optionalRoute).when(externalRoutingService).findBestNexHop(sourcePacket.getDestination());
+		ConnectorSettings connectorSettings = mock(ConnectorSettings.class);
+		when(connectorSettings.operatorAddressSafe()).thenReturn(InterledgerAddress.of(TEST_ADDRESS));
+		when(connectorSettingsSupplier.get()).thenReturn(connectorSettings);
+
+		defaultNextHopPacketMapper.getNextHopPacket(sourceAccountSettings, sourcePacket);
+	}
+
+	@Test(expected = InterledgerProtocolException.class)
+	public void getNextHopPacketSourceAccountIdMatchesNextHopAccountId() {
+		final AccountId accountId = AccountId.of(UUID.randomUUID().toString());
```

Comment:
Minor nit, but I think your indentation spacing is off. Tab-width should be 2 spaces, not 2 tabs.

```suggestion
    final AccountId accountId = AccountId.of(UUID.randomUUID().toString());
```

Reply:
Fixed (i think) as part of upcoming commit.